client/lib/upgrades/actions/free-checkout.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+import { cartItems, emptyTemporaryCart, fillInAllCartItemAttributes } from 'lib/cart-values';
+import productsListFactory from 'lib/products-list';
+import { fullCreditsPayment } from 'lib/store-transactions';
+import { submitTransaction } from './checkout';
+
+const productsList = productsListFactory();
+
+export function freeTrial( siteID, plan, onComplete ) {
```

Comment:
Yeah it probably makes more sense to name this file `free-trials.js` while it only contains a method for free trials. I wanted to be too generic. `startFreeTrial` is better also